app/api_v1/resources.py

```diff
@@ -0,0 +1,369 @@
+            id=bucketlist_id, created_by=g.user.id).first()
+        if not bucketlist:
+            return {'message': 'The bucketlist you want to insert an item to does not exists.'}, 404
+
+        if Item.query.filter_by(name=item_name, bucketlist_id=bucketlist_id).first():
+            return {'message': 'An item with the provided item name already exists.'}, 409
+        if description:
+            new_item = Item(item_name, bucketlist_id, description)
+        else:
+            new_item = Item(item_name, bucketlist_id)
+        db.session.add(new_item)
+        db.session.commit()
+        created_item = new_item
+
+        return {"successfuly created: ": marshal(new_item, item_serializer)}
+
+
+class BucketItemsApi(Resource):
+    def __init__(self):
+        self.parse = reqparse.RequestParser()
+        self.parse.add_argument('name', type=str, required=True,
+                                help='Bucket list id not provided',
+                                location='form')
+        self.parse.add_argument('description', type=str,
+                                location='form')
+        super(BucketItemsApi, self).__init__()
+
+    @auth.login_required
+    def put(self, id, item_id):
+        """
+        Edits/updates the an item with the given item_id and within the bucketlist with the given id
+
+        parameters:
+          -name
+          -description
+          -done
+          -id
+          -item_id
+
+
+        """
+        parser = reqparse.RequestParser()
+        parser.add_argument('name', type=str)
+        parser.add_argument('done', type=bool)
+        parser.add_argument('description', type=str)
+        edit_item = Item.query.filter_by(bucketlist_id=id, id=item_id).first()
+
+        if edit_item == None:
+            return {'message': 'The item you tried to edit does not exist.'}, 404
+        else:
+            args = parser.parse_args()
+            done = args['done']
+            name = args['name']
+            description = args['description']
+            if done:
+                edit_item.done = done
+            if name:
+                edit_item.name = name
+            if description:
+                edit_item.description = description
+            db.session.add(edit_item)
+            db.session.commit()
+            return {"successfully upadated: ": marshal(edit_item, item_serializer)}, 200
+
+    @auth.login_required
+    def delete(self, id, item_id):
+        """
+        Deletes the an item with the given item_id and within the bucketlist with the given id
+
+        parameters:
+          -id
+          -item_id
+
+        """
+
+        delete_item = Item.query.filter_by(
+            bucketlist_id=id, id=item_id).first()
+        if delete_item == None:
+            return {'message': 'The item you tried to delete does not exist.'}, 404
+        db.session.delete(delete_item)
+        db.session.commit()
+
+        return {'message': 'Item was successfully deleted'}, 200
```

Comment:
Had an issue with 204 not allowing the return of a message, from a little dig it turns out I can either return a 200 or a 204 for a successful delete or update, guessing I'll switch to a 204 with the tradeoff of the reply message.